packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -834,6 +834,7 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
     assert(debugCheckHasMediaQuery(context));
     assert(debugCheckHasDirectionality(context));
     final EdgeInsets padding = MediaQuery.of(context).padding;
+    final EdgeInsets viewInsets = MediaQuery.of(context).viewInsets;
```

Comment:
we should get the media query once, then read the padding and viewinsets from that.

Reply:
Done in #13434